src/glue.c

```diff
@@ -0,0 +1,163 @@
+#include "verify.h"
+#include <node_api.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+napi_status HandleError(napi_env env, struct Error *err) {
+  napi_status status;
+  napi_value error;       // The js Error object
+  napi_value errors;      // The js list of error strings
+  napi_value errorString; // Used to build the list of strings
+
+  char *msg = NULL; // used to build strings
+
+  if (err == NULL) {
+    status =
+        napi_throw_error(env, NULL, "Unknown exception verifying document");
+    VF_ERROR("received a NULL Error struct pointer\n");
+    return status;
+  }
+
+  status = napi_create_array(env, &errors);
+  if (status != napi_ok) {
+    VF_ERROR("could not create js array\n");
+    return status;
+  }
+
+  int i = 0;
+  while (err != NULL) {
+    msg = VF_err_fmt(err);
+
+    if (msg == NULL) {
+      VF_ERROR("could not format Error struct %s\n", err->reason);
+      return napi_generic_failure;
+    }
+
+    status = napi_create_string_utf8(env, msg, NAPI_AUTO_LENGTH, &errorString);
+    if (status != napi_ok) {
+      VF_ERROR("could not create js string for openssl error: %s\n", msg);
+      return status;
+    }
+
+    free(msg);
+    msg = NULL;
+
+    status = napi_set_element(env, errors, i, errorString);
+    if (status != napi_ok) {
+      VF_ERROR("could not set error %s as index %d on Errors.errors array");
+      return status;
+    }
+
+    if (err == err->next) {
+      VF_ERROR("found simple-cycle in error linked list\n");
+      break;
+    }
+    err = err->next;
+    i++;
+  }
+
+  status = napi_create_error(env, NULL, errorString, &error);
+  if (status != napi_ok) {
+    VF_ERROR("could not create js Error object\n");
+    return status;
+  }
+
+  status = napi_set_named_property(env, error, "errors", errors);
+  if (status != napi_ok) {
+    VF_ERROR("could not set js Error.errors property\n");
+    return status;
+  }
+
+  VF_err_free(err);
+
+  return napi_throw(env, error);
+}
+
+napi_value Call_VF_verify(napi_env env, napi_callback_info info) {
+  napi_value outcome = NULL;
+  napi_status status;
+  size_t argc = 3;
+  napi_value argv[argc];
+  status = napi_get_cb_info(env, info, &argc, argv, NULL, NULL);
+
+  if (status != napi_ok) {
+    napi_throw_error(env, NULL, "could not get callback information");
+    return NULL;
+  }
+
+  // Get the buffer lengths
+  size_t pubkey_l;
+  size_t document_l;
+  size_t signature_l;
+
+  uint8_t *pubkey;
+  uint8_t *document;
+  uint8_t *signature;
+
+  status = napi_get_buffer_info(env, argv[0], (void *)&pubkey, &pubkey_l);
```

Comment:
I'm not sure either, but both `(void *)&` and `(void **)&` work equally well.  `&(void*)` is a compiler error though.  Since this is getting information from a Buffer (Uint8Array) from JS, it's probably best to match that in C.